src/main/java/com/library/controller/CopyController.java

```diff
@@ -0,0 +1,56 @@
+package com.library.controller;
+
+import com.library.mapper.CopyMapper;
+import com.library.model.Copy;
+import com.library.model.dto.CopyDto;
+import com.library.service.implementation.BookServiceImplementation;
+import com.library.service.implementation.CopyServiceImplementation;
+import com.library.service.implementation.RentalServiceImplementation;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@CrossOrigin("*")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/library/copy")
+public class CopyController {
+
+    private final CopyServiceImplementation copyServiceImplementation;
+    private final BookServiceImplementation bookServiceImplementation;
+    private final RentalServiceImplementation rentalServiceImplementation;
+    private final CopyMapper copyMapper;
+
+    @PostMapping
+    public Long addCopy(@RequestParam Long bookId) {
+        bookServiceImplementation.findById(bookId);
```

Comment:
to powinno być w copy service, dodatkowo najlepiej zrobić z tego metodę np. `validateBookExists(..)`

Reply:
uprościłem to, walidację wywołuje mapper więc nie robiłem tego drugi raz w copy service